src/js/utils/hooks.js

```diff
@@ -42,3 +42,38 @@ export const useDidUpdateEffect = (onUpdate, deps) => {
         firstRef.current = true;
     }, deps);
 };
+
+export function getUrlSearchParams(key) {
+    const params = new URLSearchParams(window.location.search);
+    return params.get(key) || "";
+}
+
+export function updateUrlSearchParams(newTerm, key) {
+    if (window.location.search) {
+        const search = new URLSearchParams(window.location.search);
+        search.set(key, newTerm);
+        const newUrl = `${window.location.pathname}?${search.toString()}`;
+        window.history.replaceState({}, "", newUrl);
+    }
+}
+
+export function useUrlSearchParams({ key, defaultValue = "" }) {
+    const [value, setValue] = useState(defaultValue);
```

Comment:
I think you can remove this usage of state. Get the value from the URL which removes the possibility of desync. should also simplify the rest of the hook